docs: update aot compiler docs (#35487)
angular/angular: aio/content/guide/aot-compiler.md

Angular offers two ways to compile your application:

> * **_Just-in-Time_ (JIT)**, was used as default until Angular 8.
> * **_Ahead-of-Time_ (AOT)**, which compiles your app and libraries at build time.

Reviewer: This change looks strange. For JIT we mention when it was the default and for AOT we mention what it does :smiley:
How about:

```suggestion
* **_Just-in-Time_ (JIT)**, which compiles your app in the browser at runtime. This was the default until Angular 8.
* **_Ahead-of-Time_ (AOT)**, which compiles your app and libraries at build time. This is the default since Angular 9.
```

Reviewer 2: sure looks good to me as well

Angular offers two ways to compile your application:

> * **_Just-in-Time_ (JIT)**, was used as default till Angular 8.

Reviewer 3: ```suggestion
* **_Just-in-Time_ (JIT)**, was used as default until Angular 8.
```

Reviewer 2: Done.

* **_Ahead-of-Time_ (AOT)**, which compiles your app and libraries at build time.

> AOT compilation is the default when you run the [`ng build`](cli/build) (build only) or [`ng serve`](cli/serve)  (build and serve locally) CLI commands.

Reviewer: Doesn't the default depend on your `angular.json` config?

Reviewer 2: With new Angular 9 project, AOT is default. Also ng update also makes AOT true while updating to V9.

Reviewer: AOT is by default set to true in the `angular.json` config. This is different than saying that `ng build`/`ng serve` default to using AOT. I would make this clear in the guide (because there are people that will have their configs set to `aot: false` (for whatever reason).

Given that this guide is about AOT compilation (i.e. it is not an introductory/get started quickly guide), I think we should aim for correctness (vs glossing over under-the-hood details).

WDYT?

Reviewer 2: Makes sense George, will try to add more info, like we can change this setting from angular.json.

Reviewer: Sorry, I still find this sentence inaccurate. `ng build/serve` do not default to AOT. They use whatever is in `angular.json` (and if nothing is specified, then JIT is used afaict). It is just that the CLI generates configs with `aot: true` by default.

How about something like this (and remove the alert below):

```suggestion
When you run the [`ng build`](cli/build) (build only) or [`ng serve`](cli/serve) (build and serve locally) CLI commands, the type of compilation (JIT or AOT) depends on the value of the `aot` property in your build configuration specified in `angular.json`. By default, `aot` is set to `true` for new CLI apps.

See the [CLI command reference](cli) and [Building and serving Angular apps](guide/build) for more information.
```

Reviewer 2: Thanks, will make the changes :)
